loader/loader.c

```diff
@@ -1455,6 +1468,32 @@ static VkResult loader_scanned_icd_add(const struct loader_instance *inst, struc
         icd_tramp_list->capacity *= 2;
     }
 
+    uint32_t major_version = VK_API_VERSION_MAJOR(api_version);
+    uint32_t minor_version = VK_API_VERSION_MINOR(api_version);
+    if (interface_vers <= 4 && 1 == major_version && 0 < minor_version) {
+        loader_log(inst, VULKAN_LOADER_ERROR_BIT, 0,
+                   "loader_scanned_icd_add: Driver %s says it supports api %u.%u, but only supports loader interface %u (Policy "
+                   "#LDP_DRIVER_7)",
+                   filename, major_version, minor_version, interface_vers);
+    }
+    if (interface_vers >= 1) {
+        bool vk_eixp_not_null = (loader_platform_get_proc_address(handle, "vkEnumerateInstanceExtensionProperties") != NULL);
+        bool vk_eilp_not_null = (loader_platform_get_proc_address(handle, "vkEnumerateInstanceLayerProperties") != NULL);
+        bool vk_eiv_not_null = (loader_platform_get_proc_address(handle, "vkEnumerateInstanceVersion") != NULL);
+        bool vk_gipa_not_null = (loader_platform_get_proc_address(handle, "vkGetInstanceProcAddr") != NULL);
+        bool vk_ci_not_null = (loader_platform_get_proc_address(handle, "vkCreateInstance") != NULL);
+        bool vk_gdpa_not_null = (loader_platform_get_proc_address(handle, "vkGetDeviceProcAddr") != NULL);
+        bool vk_cd_not_null = (loader_platform_get_proc_address(handle, "vkCreateDevice") != NULL);
+
+        if (vk_eixp_not_null || vk_eilp_not_null || vk_eiv_not_null || vk_gipa_not_null || vk_ci_not_null || vk_gdpa_not_null ||
+            vk_cd_not_null) {
```

Comment:
```suggestion
        if ((loader_platform_get_proc_address(handle, "vkEnumerateInstanceExtensionProperties") != NULL)
            || (loader_platform_get_proc_address(handle, "vkEnumerateInstanceLayerProperties") != NULL) 
            || (loader_platform_get_proc_address(handle, "vkEnumerateInstanceVersion") != NULL) 
            || (loader_platform_get_proc_address(handle, "vkGetInstanceProcAddr") != NULL) 
            || (loader_platform_get_proc_address(handle, "vkCreateInstance") != NULL) 
            || (loader_platform_get_proc_address(handle, "vkGetDeviceProcAddr") != NULL) 
            || (loader_platform_get_proc_address(handle, "vkCreateDevice") != NULL)) {
```
Just a style nit.
Equivalent code (save for the short circuiting).